gameplan/extends/client.py

```diff
@@ -5,21 +5,36 @@
 import frappe
 from frappe.model.base_document import get_controller
 
-
 @frappe.whitelist()
-def get_list(doctype=None, fields=None, filters=None, order_by=None, start=0, limit=20, group_by=None, parent=None, debug=False):
-	check_permissions(doctype, parent)
-	query = frappe.qb.get_query(
-		table=doctype,
-		fields=fields,
-		filters=filters,
-		order_by=order_by,
-		offset=start,
-		limit=limit,
-		group_by=group_by,
+def get_list(doctype, **kwargs):
+	controller = get_controller(doctype)
+	if hasattr(controller, "get_list_query"):
```

Comment:
A future refactoring may consider to trigger this code path exclusively on gameplan models. Without e.g. namespacing `get_list_query` or additional gp-specific signal attributes, this logic has too a high probability of one day causing unintended side effects.

Reply:
I see that all doctypes are namespaced with `gp_` maybe that can be used upfront as a cheap string-prefix check. But these type of improvements is not part of this fix.